src/braket/ahs/driving_field.py

```diff
@@ -83,6 +84,72 @@ def detuning(self) -> Field:
         r"""Field: global detuning (:math:`\Delta(t)`). Time is in s, and value is in rad/s."""
         return self._detuning
 
+    @staticmethod
+    def from_lists(
+        times: List[float], amplitudes: List[float], detunings: List[float], phases: List[float]
+    ) -> DrivingField:
+        """
+        Builds DrivingField Hamiltonian from lists defining time evolution
+        of Hamiltonian parameters (Rabi frequency, detuning, phase).
+        The values of the parameters at each time points are global for all atoms.
+
+        Args:
+            times (List[float]): The time points of the driving field
+            amplitudes (List[float]): The values of the amplitude
+            detunings (List[float]): The values of the detuning
+            phases (List[float]): The values of the phase
+        """
+        if not (len(times) == len(amplitudes) == len(detunings) == len(phases)):
+            raise ValueError(
+                "The length of the list for times, amplitudes, detunings and phases is not equal"
+            )
+
+        amplitude = TimeSeries()
+        detuning = TimeSeries()
+        phase = TimeSeries()
+
+        for t, amplitude_value, detuning_value, phase_value in zip(
+            times, amplitudes, detunings, phases
+        ):
+            amplitude.put(t, amplitude_value)
+            detuning.put(t, detuning_value)
+            phase.put(t, phase_value)
+
+        drive = DrivingField(amplitude=amplitude, detuning=detuning, phase=phase)
+
+        return drive
+
+    def concatenate(self, other: DrivingField) -> DrivingField:
+        """Concatenate two driving fields to a single driving field
+        Args:
+            other (DrivingField): The driving field to be concatenated
+        Returns:
+            DrivingField: The concatenated driving field
+        """
+        return DrivingField(
+            amplitude=self.amplitude.time_series.concatenate(other.amplitude.time_series),
+            detuning=self.detuning.time_series.concatenate(other.detuning.time_series),
+            phase=self.phase.time_series.concatenate(other.phase.time_series),
+        )
+
+    @staticmethod
+    def concatenate_list(driving_fields: List[DrivingField]) -> DrivingField:
+        """Concatenate a list of driving fields to a single driving field
+        Args:
+            driving_fields (List[DrivingField]):
+            The list of driving field time series to be concatenated
+        Returns:
+            DrivingField: The concatenated driving field
+        """
+
+        if len(driving_fields) == 0:
+            raise ValueError("The list must not be empty.")
```

Comment:
Rather than "the list" please say "driving_fields parameter", and please add documentation in the args of the function that lets the caller know an empty list is not a valid parameter, and raises documentation that explains what we raise.

Reply:
returning empty object instead